### PR TITLE
br: fix restore full fail when restore cluster <= 6.2 using br 6.3

### LINKED ISSUE #38055: failed to restore to cluster:v4.0.14 with BR:6.3.0 and the error "[403] [PD:tso:ErrResetUserTimestamp]reset user timestamp failed, the specified ts too small than now"
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

using 6.3.0 br backup v4.0.16 data
restore it to v4.0.16 cluster

### 2. What did you expect to see? (Required)
restore successfully

### 3. What did you see instead (Required)
[2022/09/20 14:54:59.930 +00:00] [ERROR] [restore.go:58] ["failed to restore"] [error="[403] \"[PD:tso:ErrResetUserTimestamp]reset user timestamp failed, the specified ts too small than now\"\n\"success\"\n http://upstream-pd.brie-acceptance-compatiblity-tps-1236633-1-658:2379/pd/api/v1/admin/reset-ts: [BR:PD:ErrPDInvalidResponse]PD invalid response; [403] \"[PD:tso:ErrResetUserTimestamp]reset user timestamp failed, the specified ts too small than now\"\n\"success\"\n

### 4. What is your TiDB version? (Required)

v6.3.0-pre

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #38055

Problem Summary:

### What is changed and how it works?
- ignore StatusForbidden of `reset-ts` to make it compatible with cluster version <= 6.2

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - before this fix, backup and restore tidb 6.2 using br 6.3 fails
  - after this fix, success
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fengou1 — APPROVED** [review]
looks good to me

File: br/pkg/pdutil/pd.go [hunk 1]
```diff
@@ -841,7 +849,12 @@ func (p *PdController) ResetTS(ctx context.Context, ts uint64) error {
 	})
 	var err error
 	for _, addr := range p.addrs {
-		_, e := pdRequest(ctx, addr, resetTSPrefix, p.cli, http.MethodPost, bytes.NewBuffer(reqData))
+		code, _, e := pdRequestWithCode(ctx, addr, resetTSPrefix, p.cli, http.MethodPost, bytes.NewBuffer(reqData))
+		// for pd version <= 6.2, if the given ts < current ts of pd, pd returns StatusForbidden.
+		// it's not an error for br
+		if code == http.StatusForbidden {
```
Comment (3pointer) [inline br/pkg/pdutil/pd.go:855]: maybe add some logs here

Comment (joccau) [inline br/pkg/pdutil/pd.go:855]: Maybe we should test restore to cluster with 3 pds

**Review by joccau — COMMENTED** [review]
rest LGMT

File: br/pkg/pdutil/pd.go (hunk 1 above)
Comment (D3Hunter) [reply br/pkg/pdutil/pd.go:855]: pd replica doesn't matter here

**Review by 3pointer — APPROVED** [review]

**Comment by D3Hunter** [comment]
lightning ci  `lightning_various_types` failed due to this https://github.com/pingcap/tidb/issues/38092

**Comment by D3Hunter** [comment]
ci pass https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/13455/pipeline

**Pull request merged**